tools/prove_blocks.sh

```diff
@@ -27,37 +27,38 @@ IGNORE_PREVIOUS_PROOFS=$4
 echo "Proving blocks ${1}..=${2}... (Total: ${TOT_BLOCKS})"
 mkdir -p $PROOF_OUTPUT_DIR
 
-for ((i=$1; i<=$2; i++))
-do
-    echo "Proving block ${i}..."
-
-    OUT_PROOF_PATH="${PROOF_OUTPUT_DIR}/b${i}.zkproof"
-    OUT_LOG_PATH="${PROOF_OUTPUT_DIR}/b${i}.log"
-
-    if [ $IGNORE_PREVIOUS_PROOFS ]; then
-        # Set checkpoint height to previous block number
-        prev_proof_num=$((i-1))
-        PREV_PROOF_EXTRA_ARG="--checkpoint-block-number ${prev_proof_num}"
-    else
-        if [ $i -gt 1 ]; then
-            prev_proof_num=$((i-1))
-            PREV_PROOF_EXTRA_ARG="-f ${PROOF_OUTPUT_DIR}/b${prev_proof_num}.zkproof"
-        fi
+OUT_LOG_PATH="${PROOF_OUTPUT_DIR}/bs_$1..=$2.log"
+prev_proof_num=$(($1-1))
+block_numbers="$1..=$2"
+
+# If we are proving only one block, set the output log path and block_numbers accordingly.
+if [ $1 -eq $2 ]; then
+    OUT_LOG_PATH="${PROOF_OUTPUT_DIR}/b$1.log"
+    block_numbers=$1
+fi
+
+if [ $IGNORE_PREVIOUS_PROOFS ]; then
+    # Set checkpoint height to previous block number
+    PREV_PROOF_EXTRA_ARG="--checkpoint-block-number ${prev_proof_num}"
+else
+    if [ $1 -gt 1 ]; then
+        PREV_PROOF_EXTRA_ARG="-f ${PROOF_OUTPUT_DIR}/b${prev_proof_num}.zkproof"
     fi
+fi
+
+cargo r --release --bin leader -- --runtime in-memory jerigon --rpc-url "$3" --block-numbers $block_numbers --proof-output-dir $PROOF_OUTPUT_DIR $PREV_PROOF_EXTRA_ARG > $OUT_LOG_PATH 2>&1
 
-    cargo r --release --bin leader -- --runtime in-memory jerigon --rpc-url "$3" --block-number $i --proof-output-path $OUT_PROOF_PATH $PREV_PROOF_EXTRA_ARG > $OUT_LOG_PATH 2>&1
-    
-    retVal=$?
-    if [ $retVal -ne 0 ]; then
-        # Some error occured.
-        echo "Block ${i} errored. See ${OUT_LOG_PATH} for more details."
-        exit $retVal
-    else
-        # Remove the log on success if we don't want to keep it.
-        if [ $ALWAYS_WRITE_LOGS -ne 1 ]; then
-            rm $OUT_LOG_PATH
-        fi
+retVal=$?
+echo $retVal
+if [ $retVal -ne 0 ]; then
+    # Some error occured.
+    echo "Blocks $1..=$2 errored. See ${OUT_LOG_PATH} for more details."
+    exit $retVal
```

Comment:
I think we shouldn't abort early in case we're proving a range. The error log should be enough while we pursue proving other blocks.

Reply:
I don't think it is feasible with the current implementation of jerigon_main.